wrangler-core/src/main/java/co/cask/directives/parser/ParseTimestamp.java

```diff
@@ -0,0 +1,94 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package co.cask.directives.parser;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.wrangler.api.Arguments;
+import co.cask.wrangler.api.Directive;
+import co.cask.wrangler.api.DirectiveExecutionException;
+import co.cask.wrangler.api.DirectiveParseException;
+import co.cask.wrangler.api.ErrorRowException;
+import co.cask.wrangler.api.ExecutorContext;
+import co.cask.wrangler.api.Row;
+import co.cask.wrangler.api.annotations.Categories;
+import co.cask.wrangler.api.parser.ColumnName;
+import co.cask.wrangler.api.parser.TokenType;
+import co.cask.wrangler.api.parser.UsageDefinition;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+/**
+ * A Executor to parse timestamp as date.
+ */
+@Plugin(type = Directive.Type)
+@Name("parse-timestamp-as-date")
+@Categories(categories = {"parser", "date"})
+@Description("Parses column values representing unix timestamp in millis as date.")
+public class ParseTimestamp implements Directive {
+  public static final String NAME = "parse-timestamp-as-date";
+  private String column;
+
+  @Override
+  public UsageDefinition define() {
+    UsageDefinition.Builder builder = UsageDefinition.builder(NAME);
+    builder.define("column", TokenType.COLUMN_NAME);
+    return builder.build();
+  }
+
+  @Override
+  public void initialize(Arguments args) throws DirectiveParseException {
+    this.column = ((ColumnName) args.value("column")).value();
+  }
+
+  @Override
+  public void destroy() {
+    // no-op
+  }
+
+  @Override
+  public List<Row> execute(List<Row> rows, ExecutorContext context)
+    throws DirectiveExecutionException, ErrorRowException {
+    for (Row row : rows) {
+      int idx = row.find(column);
+      if (idx != -1) {
+        Object object = row.getValue(idx);
+        // If the data in the cell is null or is already of date format, then
+        // continue to next row.
+        if (object == null || object instanceof ZonedDateTime) {
+          continue;
+        }
+        if (object instanceof Long) {
+          Instant i = Instant.ofEpochMilli((Long) object);
+          ZonedDateTime zonedDateTime = ZonedDateTime.ofInstant(i, ZoneId.ofOffset("UTC", ZoneOffset.UTC));
+          row.setValue(idx, zonedDateTime);
+        } else {
```

Comment:
'parse' kind of implies it would work on strings too. I think it makes sense to try and parse a string as a long.